flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -686,12 +688,25 @@ public CompletableFuture<Acknowledge> submitTask(
             PartitionProducerStateChecker partitionStateChecker =
                     jobManagerConnection.getPartitionStateChecker();
 
+            // Configuration from application will override the one from env.
+            boolean envChangelogEnabled =
+                    taskManagerConfiguration
+                            .getConfiguration()
```

Comment:
How about passing cluster and job configurations to `TaskExecutorLocalStateStoresManager.localStateStoreForSubtask`
and parsing there?

nit: I'd inline `envChangelogEnabled` but that's a matter of taste :)

Reply:
I meant something like this:
```
// in TaskExecutor
final TaskLocalStateStore localStateStore =
    localStateStoresManager.localStateStoreForSub
        jobId,
        tdd.getAllocationId(),
        taskInformation.getJobVertexId(),
        tdd.getSubtaskIndex(),
        taskManagerConfiguration.getConfigura
        jobInformation.getJobConfiguration())

// TaskExecutorLocalStateStoresManager.localStateStoreForSubtask
    ...
    boolean changelogEnabled =
        jobConfiguration
            .getOptional(
                StateChangelogOptionsInternal
                    .ENABLE_CHANGE_LOG_FOR_APPLICATION)
            .orElse(
                clusterConfiguration.getBoolean(
                    StateChangelogOptions.ENABLE_STATE_CHANGE_LOG));
```